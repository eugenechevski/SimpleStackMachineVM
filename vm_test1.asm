	# $Id: vm_test1.asm,v 1.6 2024/07/29 02:01:32 leavens Exp $
	.text start
start:	STRA
	SRI $sp, 2         # allocate 2 words on the stack
	ADDI $sp, 1, 1     # put 1 in memory[$sp+1]
	ADD $gp, 2, $sp, 1 # put 1 in temp
	ADD $sp, 0, $gp, 2 # put 1 in stack top
	SUB $gp, 2, $gp, 2 # zero out temp
	ADDI $sp, 0, 10    # add 10 to stack top, should be 11
	SRI $sp, 1         # allocate a word on the stack
	NOTR
	PCH $gp, 0         # print y
	PCH $gp, 3         # print nl
	EXIT 0
	.data 1024
	CHAR y = 'y'
	CHAR n = 'n'
	WORD temp    # not initialized
	CHAR nl = '\n'
	.stack 4096
	.end
	
