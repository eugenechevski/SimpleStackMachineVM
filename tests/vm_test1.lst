Address Instruction
     0: STRA 
     1: SRI $sp, 2
     2: ADDI $sp, 1, 1
     3: ADD $gp, 2, $sp, 1
     4: ADD $sp, 0, $gp, 2
     5: SUB $gp, 2, $gp, 2
     6: ADDI $sp, 0, 10
     7: SRI $sp, 1
     8: NOTR 
     9: PCH $gp, 0
    10: PCH $gp, 3
    11: EXIT 0
    1024: 121	    1025: 110	    1026: 0	    1027: 10	    1028: 0	
        ...     
