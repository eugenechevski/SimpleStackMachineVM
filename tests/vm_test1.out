      PC: 0
GPR[$gp]: 1024 	GPR[$sp]: 4096 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 0	    1027: 10	    1028: 0	
        ...     
    4096: 0	

==>      0: STRA 
      PC: 1
GPR[$gp]: 1024 	GPR[$sp]: 4096 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 0	    1027: 10	    1028: 0	
        ...     
    4096: 0	

==>      1: SRI $sp, 2
      PC: 2
GPR[$gp]: 1024 	GPR[$sp]: 4094 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 0	    1027: 10	    1028: 0	
        ...     
    4094: 0	        ...     

==>      2: ADDI $sp, 1, 1
      PC: 3
GPR[$gp]: 1024 	GPR[$sp]: 4094 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 0	    1027: 10	    1028: 0	
        ...     
    4094: 0	    4095: 1	    4096: 0	

==>      3: ADD $gp, 2, $sp, 1
      PC: 4
GPR[$gp]: 1024 	GPR[$sp]: 4094 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 1	    1027: 10	    1028: 0	
        ...     
    4094: 0	    4095: 1	    4096: 0	

==>      4: ADD $sp, 0, $gp, 2
      PC: 5
GPR[$gp]: 1024 	GPR[$sp]: 4094 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 1	    1027: 10	    1028: 0	
        ...     
    4094: 1	    4095: 1	    4096: 0	

==>      5: SUB $gp, 2, $gp, 2
      PC: 6
GPR[$gp]: 1024 	GPR[$sp]: 4094 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 0	    1027: 10	    1028: 0	
        ...     
    4094: 1	    4095: 1	    4096: 0	

==>      6: ADDI $sp, 0, 10
      PC: 7
GPR[$gp]: 1024 	GPR[$sp]: 4094 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 0	    1027: 10	    1028: 0	
        ...     
    4094: 11	    4095: 1	    4096: 0	

==>      7: SRI $sp, 1
      PC: 8
GPR[$gp]: 1024 	GPR[$sp]: 4093 	GPR[$fp]: 4096 	GPR[$r3]: 0    	GPR[$r4]: 0    
GPR[$r5]: 0    	GPR[$r6]: 0    	GPR[$ra]: 0    
    1024: 121	    1025: 110	    1026: 0	    1027: 10	    1028: 0	
        ...     
    4093: 0	    4094: 11	    4095: 1	    4096: 0	

==>      8: NOTR 
y
